File: JanperCab/5-JanperCab/ClientApp/src/app/_models/process/ProcessPressingDto.ts
import { MachinePresserDto } from '../machine/MachinePresserDto';
import { ProcessDto } from './ProcessDto';

export class ProcessPressingDto extends ProcessDto {
  machineId: number;

  machinePresser: MachinePresserDto;

  getStatus(): string {
    status = this.startTime && this.endTime ? 'Pressed' : 'Pressing';

    if (this.machinePresser) {
      status += ` - ${this.machinePresser.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/ProcessPressingDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class ProcessPressingDto : ProcessDto
    {
        public int? MachineId { get; set; }

        public MachinePresserDto MachinePresser { get; set; }
    }
}